core/src/main/kotlin/org/evomaster/core/search/gene/OptionalGene.kt

```diff
@@ -52,7 +52,7 @@ class OptionalGene(name: String,
     }
 
     override fun isMutable(): Boolean {
-        return selectable
+        return selectable && gene.isMutable()
     }
```

Comment:
i do not think this one is correct.
if an OptionalGene X contains an immutable gene Y, still we can always mutate X, as we can flip its on/off status.
or is there any other issue i m missing here?

Reply:
then if an OptionalGene X contains an immutable gene Y, only `isActive` can be mutated?

Reply:
yes, if Y is immutable, the only mutation allowed is to flip `isActive`, which depends on `selectable`

Reply:
look at how `standardMutation` is implemented. but if that was not clear from the code, maybe it could make sense to add some explanation about it